common/src/main/java/bio/terra/catalog/service/DatasetService.java

```diff
@@ -170,10 +175,13 @@ public void deleteMetadata(DatasetId datasetId) {
   public String getMetadata(DatasetId datasetId) {
     var dataset = datasetDao.retrieve(datasetId);
     ensureActionPermission(dataset, SamAction.READ_ANY_METADATA);
-    return new DatasetResponse(
-            dataset, getService(dataset.storageSystem()).getDataset(dataset.storageSourceId()))
-        .convertToObject()
-        .toString();
+    StorageSystemInformation information;
+    try {
+      information = getService(dataset.storageSystem()).getDataset(dataset.storageSourceId());
+    } catch (ForbiddenException e) {
```

Comment:
Maybe I'm missing something, but could a non-admin caller legitimately not have access to the underlying dataset and find themselves in this catch block, and illegitimately get access that way?

Reply:
The thing to keep in mind with the catalog is that the catalog's data (a catalog entry) is separate from the underlying object's data, which a catalog entry represents. Everything in the catalog (all catalog entries) are visible to, and editable by, a catalog admin. The underlying object's data (which this code calls the storage system data) may or may not be visible to a catalog admin user.

Since the catalog itself has no special access to any data in an underlying system (like TDR), the catalog service itself can never show a user data that they would't be able to access normally (e.g., via swagger or TDR UI). The only thing the catalog can "expose" is data within the catalog itself.

The underlying storage system permissions are used in the catalog UI to indicate that an operation is possible, such as exporting a snapshot to a workspace. But when it comes time to do that operation, it's done using the current user's credentials, so if it turns out that they don't really have permission to do it, TDR (and the import service) won't let it happen.

Reply:
Okay, fair enough.  I'm convinced that this doesn't pose a permissions issue but the use of `ADMIN_INFORMATION` as the fallback variable name is still a bit confusing to me.  That could be because I'm less familiar with the repo though.

Reply:
> the use of `ADMIN_INFORMATION` as the fallback variable name is still a bit confusing to me

That's a good point, I've renamed it to `DEFAULT_INFORMATION` which should be clearer.